tests/profiling/collector/test_stack.py

```diff
@@ -574,17 +576,10 @@ def test_resource_not_collected(monkeypatch, tracer):
         resource = str(uuid.uuid4())
         span_type = str(uuid.uuid4())
         span = tracer.start_span("foobar", activate=True, resource=resource, span_type=span_type)
-        # This test will run forever if it fails. Don't make it fail.
-        while True:
-            try:
-                event = collector.recorder.events[stack_event.StackSampleEvent].pop()
-            except IndexError:
-                # No event left or no event yet
-                continue
-            if span.span_id == event.span_id:
-                assert event.trace_resource_container is None
-                assert event.trace_type == span_type
-                break
+        event = wait_for_event(collector, lambda e: span.span_id == e.span_id)
```

Comment:
Should we also update  `test_collect_multiple_span_id`? (on line 587)

Reply:
Ah, yes! I've missed that, well spotted!